validation-test/compiler_crashers_2_fixed/rdar64992293.swift

```diff
@@ -0,0 +1,12 @@
+// RUN: not %target-swift-frontend -typecheck %s
+
+public protocol SomeProtocol {}
+
+public struct Impl<Param>: SomeProtocol where Param: SomeProtocol {}
+
+public struct Wrapper<Content> where Content: SomeProtocol {}
+
+public extension Wrapper where Content == Impl<WrapperParam> {
+  typealias WrapperParam = SomeProtocol
+}
+
```

Comment:
Slightly more reduced:
```swift
protocol SomeProtocol {}
struct G<T> {}

struct Wrapper<Content: SomeProtocol> {}

extension Wrapper where Content == G<Wrapper.WrapperParam> {
  typealias WrapperParam = Never
}
```

Reply:
The original test case came from a radar. I like to keep those mostly as-is, in case they uncover more bugs in the future.